### PR TITLE
feature(website): storybook docusaurus integration

### LINKED ISSUE #7438: integrate storybook using the result of the investigation
### Is your enhancement related to a problem? Please describe

implement chosen path from https://github.com/containers/podman-desktop/issues/7437

### Describe the solution you'd like

after the study, use result of https://github.com/containers/podman-desktop/issues/7437

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

First draft of the Storybook integration this pull request contains the following

#### A `Docusaurus` plugin 

This plugin, which is located in `website/storybook.js` is responsible of generating the sidebar and populate the docusaurus build folder with the storybook sources.

#### A react page

Once the plugin has generated a sidebar under the docusaurus format, a custom react page will use it to create a page located at `/storybook`.

### Screenshot / video of UI

TODO

### What issues does this PR fix or reference?

Fixes https://github.com/containers/podman-desktop/issues/7438

### How to test this PR?

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: website/storybook.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+const fs = require('node:fs');
```
Comment (benoitf) [inline website/storybook.js]: should it be using typescript rather than js as it seems docusaurus can use typescript for configuration files

Comment (axel7083) [reply website/storybook.js]: Yeah, maybe, I will try,  but this is a plugin so need to do some testing

Comment (axel7083) [reply website/storybook.js]: ✅ done

**Comment by axel7083** [comment]
Here is a demo https://6672bf678132600ac54716c3--podman-desktop-pr.netlify.app/storybook?id=example-button--docs

## ℹ️ Special note

The sidebar is very basic on purpose, I will need to improve it, but as it will probably be a lot of work, it will be made in a follow-up PR

**Comment by benoitf** [comment]
it looks like vale is broken

**Comment by deboer-tim** [comment]
Odd flashing when toggling between pages, but I assume that can mostly be fixed by having consistent light/dark mode?

Also noticed some Storybook issues like inprogress not quite working, enums, hard to clear values, but those are separate.

**Comment by axel7083** [comment]
> Odd flashing when toggling between pages, but I assume that can mostly be fixed by having consistent light/dark mode?

yeah this is very complicated, as the default value of the iframe is set to light mode, and the docusaurus page will send the theme value when the page is rendered. We can probably improve that later, but for now this little flickering is here.

> Also noticed some Storybook issues like inprogress not quite working, enums, hard to clear values, but those are separate.

Yeah, because all values are extracted from reflection on svelte component, we can improve those later

**Review by benoitf — APPROVED** [review]
you remove the job for storybook as now it's part of the website ?

**Comment by axel7083** [comment]
> you remove the job for storybook as now it's part of the website ?

Yes, sorry I forgot to mention it

**Comment by benoitf** [comment]
I clicked on the rebase button to get fixes for argos/vale

**Comment by benoitf** [comment]
looks like it's all ✅

**Pull request merged**